os_migrate/roles/conversion_host/tasks/main.yml

```diff
@@ -117,3 +117,20 @@
     ca_cert: "{{ os_migrate_conversion_ca_cert|default(omit) }}"
     client_cert: "{{ os_migrate_conversion_client_cert|default(omit) }}"
     client_key: "{{ os_migrate_conversion_client_key|default(omit) }}"
+
+- name: create the conversion host floating IP
+  openstack.cloud.floating_ip:
+    server: "{{ os_migrate_conversion_host_name }}"
+    network: "{{ os_migrate_conversion_external_network_name }}"
+    nat_destination: "{{ os_migrate_conversion_net_name }}"
+    state: present
+    wait: yes
+    timeout: "{{ os_migrate_conversion_host_deploy_timeout }}"
+    reuse: yes
+    auth: "{{ os_migrate_conversion_auth }}"
+    auth_type: "{{ os_migrate_conversion_auth_type|default(omit) }}"
+    region_name: "{{ os_migrate_conversion_region_name|default(omit) }}"
+    validate_certs: "{{ os_migrate_conversion_validate_certs|default(omit) }}"
+    ca_cert: "{{ os_migrate_conversion_ca_cert|default(omit) }}"
+    client_cert: "{{ os_migrate_conversion_client_cert|default(omit) }}"
+    client_key: "{{ os_migrate_conversion_client_key|default(omit) }}"
```

Comment:
If we go for the manual floating IPs, this task would then probably have `when: os_migrate_conversion_floating_ips is defined and os_migrate_conversion_floating_ips not is string`.

(I think defaulting to `omit` on the `src/dst`-specific variables generates a value which is some specially formed string, so we need to check that the value is not a string. I wanted to check whether the value is list, there are `is iterable` and `is sequence` tests in Jinja, however they both match both a string and a list. There does not seem to be a Jinja test to specifically check for list only, based on my googling. :shrug:

Reply:
Reference: https://jon.sprig.gs/blog/post/1801

Reply:
Yeah the problem is that  floating_ip_address is a string so we should default it to something not a string, like false

Reply:
> If we go for the manual floating IPs, this task would then probably have `when: os_migrate_conversion_floating_ips is defined and os_migrate_conversion_floating_ips not is string`.
> 
> (I think defaulting to `omit` on the `src/dst`-specific variables generates a value which is some specially formed string, so we need to check that the value is not a string. I wanted to check whether the value is list, there are `is iterable` and `is sequence` tests in Jinja, however they both match both a string and a list. There does not seem to be a Jinja test to specifically check for list only, based on my googling. 🤷 )

I'm afraid that `is defined` will always pass and the parameter must be a string, in this case if we default it to false, we can check whether if the variable is a string or not as it should be a string (https://docs.ansible.com/ansible/latest/collections/openstack/cloud/floating_ip_module.html#parameter-floating_ip_address)